src/core/jsonpointer/include/sourcemeta/core/jsonpointer.h

```diff
@@ -52,6 +52,12 @@ const WeakPointer empty_weak_pointer;
 /// A JSON Pointer with unresolved wildcards
 using PointerTemplate = GenericPointerTemplate<Pointer>;
 
+// overloaded functions
+auto get(const JSON &&document, const Pointer &pointer)
```

Comment:
Also, could this be like this:

```suggestion
auto get(JSON &&document, const Pointer &pointer)
```

Reply:
I don't think so, because the check talks about returning const ref parameter as const ref.
I will try to test it though